### PR TITLE
Option to configure displaying of copyright in the footer

### LINKED ISSUE #502: Add an option to be able to enable/disable the display of the copyright "Ecommerce software by PrestaShop" in the footer
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5297).

- _**Reporter:**_ marion_francois
- _**Created at:**_ Thu, 12 Apr 2018 09:10:49 +0200

<p>Before in 1.6 version we had the CMS block module allowing to configure links in the footer and display or not the copyright "powered by PrestaShop".</p>

<p>This module has been replaced in 1.7 by the tab Design > Link Widget.<br/>
But we can't anymore choosing to display or not the copyright.</p>

<p>So, if I want to remove it, I have to edit the file in templates/_partials/footer.tpl, which isn't easy for a merchant</p>
        
## Specs

1. In Shop parameters > General page, under "Display best sellers", add a toggle "Display PrestaShop copyright in the footer" Yes/no, set on Yes by default.
2. If the toggle is set on No, then the following sentence will not be displayed:
capture d'écran_2924

### LINKED ISSUE #9674: [BOOM-5297] Link widget copyright
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5297).

- _**Reporter:**_ marion_francois
- _**Created at:**_ Thu, 12 Apr 2018 09:10:49 +0200

<p>Before in 1.6 version we had the CMS block module allowing to configure links in the footer and display or not the copyright "powered by PrestaShop".</p>

<p>This module has been replaced in 1.7 by the tab Design > Link Widget.<br/>
But we can't anymore choosing to display or not the copyright.</p>

<p>So, if I want to remove it, I have to edit the file in templates/_partials/footer.tpl, which isn't easy for a merchant</p>
        
* How to reproduce the issue ?

N/A

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This is a new option in the back office to allow merchants to hide PrestaShop copyright information from the footer
| Type?             | new feature
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Partially Fixes PrestaShop/hummingbird#502. - PR for the theme and autoupgrade module are coming
| How to test?      | Instruction below
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

Install PrestaShop, use `{$configuration|dump}` somewhere in `footer.tpl`, you should see a configuration related to copyrights `display_footer_copyright`, hiding it will be implemented in the theme repository

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27963)

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I don't think this one should be integrated in the Core as the theme can remove the copyright itself, or emails theme. Also, it can be done by a module if needed.

**Comment by kpodemski** [comment]
@PierreRambaud Issue has been validated and accepted by PM. AFAIK there are no plans to have a module with a dedicated setting for a theme, especially classic.

**Comment by PierreRambaud** [comment]
> @PierreRambaud Issue has been validated and accepted by PM. AFAIK there are no plans to have a module with a dedicated setting for a theme, especially classic.

Adding configuration per theme using the theme.yml is easier to maintain et really better :(

**Comment by kpodemski** [comment]
@PierreRambaud 

Agree, but this configuration option is for the merchant :-) And theme authors can override this option with their .yml.

I would like to take this opportunity to ask what now. The Issue was discussed, the PM accepted it. I invested time, and it turns out that the maintainer disagrees with the PM. What now? It's a smaller contribution, and I'm not a first-timer here, but what about more extensive topics and new contributors? This is not encouraging further contributions.

cc @PrestaShop/prestashop-maintainers @PrestaShop/product-team

**Comment by PierreRambaud** [comment]
> @PierreRambaud
> 
> Agree, but this configuration option is for the merchant :-) And theme authors can override this option with their .yml.
> 
> I would like to take this opportunity to ask what now. The Issue was discussed, the PM accepted it. I invested time, and it turns out that the maintainer disagrees with the PM. What now? It's a smaller contribution, and I'm not a first-timer here, but what about more extensive topics and new contributors? This is not encouraging further contributions.
> 
> cc @PrestaShop/prestashop-maintainers @PrestaShop/product-team

We can't follow all issues, this is not a reason why I have to approve all changes only because PM accepted it :thinking: 
Also, something that has been approved months ago doesn't mean it makes sense to integrate it now.
Anyway, I'll not approve this change as I don't think it's the right way to add this kind of feature :)

**Comment by kpodemski** [comment]
This way, the validated issue has to be discussed again with every maintainer before starting work to make sure that it's worth investing time 🤔 I'm not sure if this is the proper process. I'm not sure what's the point of waiting for PMs in so many issues if, in the end, it's worthless.

**Comment by SharakPL** [comment]
@kpodemski why `display_footer_copyright`? IMO it's too specific. Should be simply `display_copyright`

**Comment by kpodemski** [comment]
I need help with this one @PrestaShop/prestashop-maintainers @PrestaShop/product-team 

If the issue is outdated and not really "Ready", I suggest improving it so I can continue or delete it if this feature is no longer needed.

**Comment by matks** [comment]
@kpodemski did a discussion happen and solve this issue or is there still some uncertainty?

**Comment by kpodemski** [comment]
@matks I still don't know what to do with this one, and my question about the process is still unanswered.

**Comment by PierreRambaud** [comment]
> @matks I still don't know what to do with this one, and my question about the process is still unanswered.

You asked it only 2 days ago, I'm pretty sure you can wait a little bit more :p

**Comment by MatShir** [comment]
I don't see why this PR is blocked if there is no module planned.
There are other features should also be in a module as well.

Capture d’écran 2022-03-24 à 19 21 42
Capture d’écran 2022-03-24 à 19 20 32

Maybe @eternoendless can clarify what are the requirement for the project ?

**Comment by Hlavtox** [comment]
I don't think core should be cluttered with configurations like these. We could then have a million options built in - show this there, show this on left, show this on right, show purple elephant over there... meh. Deleting copyright from the theme is like 15 second work.

We had a call with @NeOMakinG today and we discussed something like an automatic theme configuration page, that could load theme options from YML. Something like Wordpress theme customization. No core options needed, no theme configurator modules needed. https://github.com/PrestaShop/hummingbird/issues/218

**Comment by kpodemski** [comment]
I like the idea @Hlavtox but:

- yml is not suitable for it, it should be PHP file
- simple solution is probably not enough,because most themes will come up with their own configurators anyway, to have fields that depends on each other, etc.

of course, we could have something that powerful as theme customizer, but with so many things to do, (definitely more important for ecommerce software) I'm not sure if anyone will have time for this task 😐

> Deleting copyright from the theme is like 15 second work.

not really, for example, you need child-theme to keep your changes in case of update - and as this is a common request, there was an idea to make it configurable in a most simple way possible 😁

**Comment by mparvazi** [comment]
@Hlavtox 

The config rules in theme.yml must be converted to a css file (e.g. override.css)
But the styling rules have been implemented by scss, so it would be error prone.

Currently we have this problem in RTL for BO. The rules for fixing RTL issues are in css (theme.rtlfix) but BO theme styled by scss.

I mean, something like this makes it harder for theme maintainability.

**Review by Progi1984 — CHANGES_REQUESTED** [review]
I don't think this should be embed in the Core. It should be embed in a module in the theme.

**Comment by Hlavtox** [comment]
There would be sooooo many things the customers would like to configure in a theme.... The copyright, quick view, manufacturer on product page, availability on product page, add to cart button on product lists and million other options. :-)

A configurator for this bundled with theme would be nice on one side (as 3rd party themes do it), but on the other side, it's always additional performance loss and complexity. Every configuration is just adding DB queries and if/else statements. With basic HTML skills, you can edit the theme to your liking, remove crap you don't need and keep the performance, maybe even improve it by not executing uneeded code and conditions.

**Comment by kpodemski** [comment]
@Hlavtox 

Of course, but I don't see questions about this as often as I see questions about removing copyrights. That is a PR to solve a common issue/request, it's probably an overkill to create a module with only one option to configure footer copyrights :-)

Anyway, I'll let wait for what's next because it seems there's a gap in the process as in this PR @PrestaShop/product-team don't agree with @PrestaShop/prestashop-maintainers and I'm not sure what now ;-)

**Comment by alfonsolr09** [comment]
I think that it is not a feature that I specifically need (since it can be removed with css for example in a simple way) but it is a useful feature for someone who wants to set up a store without knowledge

**Pull request closed without merging**
